### PR TITLE
Add service dependency startup logic

### PR DESCRIPTION
**Pull request by weichou1229**
https://github.com/edgexfoundry/device-sdk-go/issues/7

Signed-off-by: Bruce <weichou1229@gmail.com>

### CODE REVIEW

File: service_client_initializer.go [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+	params.ServiceKey = svc.c.Clients[ClientData].Name
+	params.Path = dataPath
+	params.UseRegistry = svc.useRegistry
+	params.Url = dataURL
+
+	svc.ec = coredata.NewEventClient(params, types.Endpoint{})
+
+	params.Path = v1Valuedescriptor
+	params.Url = dataAddr + dataPath
```
Comment (jduranf) [inline service_client_initializer.go]: PR #39 isn't applied, dataPath should be changed by params.Path

File: service_client_initializer.go [hunk 2]
```diff
@@ -0,0 +1,221 @@
+package device
```
Comment (jduranf) [inline service_client_initializer.go]: snake_case isn't used in the other SDK files, so maybe the file name should be changed
There is a related issue in [edgex-go](https://github.com/edgexfoundry/edgex-go/issues/535)

Comment (weichou1229) [reply service_client_initializer.go]: I change it to clientinitializer

File: service_client_initializer.go (hunk 1 above)
Comment (weichou1229) [reply service_client_initializer.go]: Yes, you're right. It didn't pass the v1Valuedescriptor value.

**Pull request merged**